tests/blas/nodes/axpy_test.py

```diff
@@ -106,8 +113,43 @@ def test_pure():
 
 
 def fpga_graph(veclen, dtype, test_case, expansion):
-    sdfg = pure_graph(veclen, dtype, test_case, expansion)
+    sdfg = pure_graph(veclen, dtype, test_case, expansion, False)
+    sdfg.apply_transformations_repeated([FPGATransformSDFG, InlineSDFG])
+    return sdfg
+
+
+def fpga_hbm_graph(veclen, dtype, expansion):
+    sdfg = pure_graph(veclen, dtype, "fpga_hbm", expansion, False)
+
+    banks_per_array = 2
+    per_array_size = sdfg.arrays["x"].shape[0] / banks_per_array
+    utils.update_array_shape(sdfg, "x", [banks_per_array, per_array_size])
+    utils.update_array_shape(sdfg, "y", [banks_per_array, per_array_size])
+    sdfg.arrays["x"].location["memorytype"] = "HBM"
+    sdfg.arrays["y"].location["memorytype"] = "HBM"
+    sdfg.arrays["x"].location["bank"] = f"0:{banks_per_array}"
+    sdfg.arrays["y"].location["bank"] = f"{banks_per_array}:{2*banks_per_array}"
+    state = sdfg.states()[0]
+    for node in state:
+        if isinstance(node, nodes.AccessNode):
+            utils.update_path_subsets(
+                state, node,
+                subsets.Range.from_string(
+                    f"0:{banks_per_array}, 0:{per_array_size}"))
+    libnode = list(
+        filter(lambda x: isinstance(x, nodes.LibraryNode), state.nodes()))[0]
+    libnode.n = dace.symbolic.pystr_to_symbolic(f"n/{banks_per_array}")
     sdfg.apply_transformations_repeated([FPGATransformSDFG, InlineSDFG])
+    sdfg.expand_library_nodes()
+    utils.update_array_shape(sdfg, "x", [per_array_size * banks_per_array])
+    utils.update_array_shape(sdfg, "y", [per_array_size * banks_per_array])
+    sdfg.arrays["x"].storage = dtypes.StorageType.CPU_Heap
+    sdfg.arrays["y"].storage = dtypes.StorageType.CPU_Heap
+    for xform in optimizer.Optimizer(sdfg).get_pattern_matches(
+            patterns=[hbm_copy_transform.HbmCopyTransform]):
+        xform.apply(sdfg)
```

Comment:
(comment for all the transformations) Did you know that you can do `HbmCopyTransform.apply_to(sdfg)` without creating a match object first?

Reply:
No, but learn't about that a few days ago ;)